param/parameterized.py

```diff
@@ -1957,45 +2164,59 @@ def outputs(self_):
                     outputs[name] = (otype, method, idx)
         return outputs
 
-
-    def _spec_to_obj(self_,spec):
-        # TODO: when we decide on spec, this method should be
-        # rewritten
-
+    def _spec_to_obj(self_, spec, dynamic=True):
```

Comment:
Definitely needs a docstring to say what it is supposed to be doing for such a tricky method!